From 935003883e02a8a2af79ccc483ad4f9e9d2e50c7 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 13 Oct 2020 19:19:10 -0700
Subject: [PATCH] parser: Add support for CAP_CHECKPOINT_RESTORE

Linux 5.9 added CAP_CHECKPOINT_RESTORE add it to the set of supported
capabilities.

MR: https://gitlab.com/apparmor/apparmor/-/merge_requests/654
Signed-off-by: John Johansen <john.johansen@canonical.com>
Acked-by: Seth Arnold <seth.arnold@canonical.com>
(cherry picked from commit 644a473971df4e18555e97fa36bafd89459c4717)
Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 parser/base_cap_names.h | 2 ++
 parser/capability.h     | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/parser/base_cap_names.h b/parser/base_cap_names.h
index 6886ed99..9f922c22 100644
--- a/parser/base_cap_names.h
+++ b/parser/base_cap_names.h
@@ -8,6 +8,8 @@
 
 {"bpf", CAP_BPF, CAP_SYS_ADMIN, CAPFLAG_BASE_FEATURE},
 
+{"checkpoint_restore", CAP_CHECKPOINT_RESTORE, CAP_SYS_ADMIN, CAPFLAG_BASE_FEATURE},
+
 {"chown", CAP_CHOWN, NO_BACKMAP_CAP, CAPFLAG_BASE_FEATURE},
 
 {"dac_override", CAP_DAC_OVERRIDE, NO_BACKMAP_CAP, CAPFLAG_BASE_FEATURE},
diff --git a/parser/capability.h b/parser/capability.h
index 7d1b7a29..23edf7c6 100644
--- a/parser/capability.h
+++ b/parser/capability.h
@@ -29,6 +29,10 @@
 #define CAP_BPF 39
 #endif
 
+#ifndef CAP_CHECKPOINT_RESTORE
+#define CAP_CHECKPOINT_RESTORE 40
+#endif
+
 typedef enum capability_flags {
 	CAPFLAGS_CLEAR = 0,
 	CAPFLAG_BASE_FEATURE = 1,
-- 
2.30.2

